### PR TITLE
Fix: Dialog Modal from Advanced Settings Hiding on clicking Add Images

### LINKED ISSUE #1156: Fix: Images Modal not hiding / closing when selecting "Add Images"
## This is:

- Bug

## Specifications

  - version:
  - OS:
  - Browser:

## Expected Behavior

When drag & drop an image gallery page element, and click Add Images button from the modal, the upload view appears, the modal should be closed or hidden temporarely.

## Actual Behavior

The modal is staying open while the page in the background loads to the images page. Only when we click outside the modal, or on the save button, the modal closes.

## Steps to Reproduce the Problem

1. Open a page in the editor, and select "add element" in the side nav ('+' icon within circle).
2. Add an image gallery page element by click and dragging into editor page, which should pop up a modal.
3. If it doesn't open, make sure the section within the editor is opened before dragging it in, if not try clicking an element first.
3. Once Modal is open, just click the "Add Images" button. As you can see the images page loads in the background but the modal is not closing.

## Detailed Description

Have the Add Images button close or hide the modal as well. Will look at other ways it is closing.

This issue was originally highlighted within issue [#1113](https://github.com/webiny/webiny-js/issues/1113), but it had no relevance to the main issue.

### PR DESCRIPTION
**Pull request by TommyJackson85**
## Related Issue

Closes #1156 

## Your solution

UPDATE:
As @Pavel910 noted in the comments, the ```zIndex``` of ```OverlayLayoutWrapper``` just needed to be bumped up to 50, and the modal only needs to be hidden.
The modal is now always there just underneath the image selector.
  
## How Has This Been Tested?
Tested manually on Chrome browser.

## Screenshots (if relevant):

UPDATE:
With the zIndex up to 50, the following happens:

Dragging the Image Gallery button into the editor...
Screenshot 2020-07-26 at 17 06 06

Opens popup.
Screenshot 2020-07-26 at 17 06 16

Then clicking Add Images pushes the image selector layout above the modal. (Am I right?)
Screenshot 2020-07-26 at 17 06 36

Proceeding with the selected 2 images hides or removes the image selector to show the modal again.
Screenshot 2020-07-26 at 17 06 46

### CODE REVIEW

**Comment by TommyJackson85** [comment]
Marked files (with console logs) where potential file changes may be done.

**Comment by TommyJackson85** [comment]
Set up the closeDialog function. 
@Pavel910 I am looking for a code review but I know myself this doesn't work. I am just looking for some guidance on how I might go about it.

**Comment by Pavel910** [comment]
Hey @TommyJackson85, actually you don't need to close the dialog - that would, in fact, be completely wrong. The problem is with `z-index` of File Manager overlay. That's the only thing that needs fixing: 
In here `packages/app-admin/src/components/OverlayLayout/OverlayLayout.tsx:16` - this `zIndex` needs to be bumped up to about `50`. We always want FileManager to be at the very very top when opened.

Once this is done, the FM overlay will properly open above the AdvancedSettings dialog, and once you select images, FM will close, and you'll be back at your dialog.

**Comment by TommyJackson85** [comment]
Thanks Pavel! I’ll edit both the issue and pr tasks for this issue.

**Comment by TommyJackson85** [comment]
@Pavel910 Well that was much simpler change needed than I thought, haha! Modal is now hidden on clicking Add Images showing the image selector in full. Furthermore, when selecting the images the modal pops back up showing the selected images, while also loading back to the initial page being edited.

Would you like me to remove that long comment next to the ```zIndex``` ?

File: packages/app-admin/src/components/OverlayLayout/OverlayLayout.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ const OverlayLayoutWrapper = styled("div")({
     width: "100%",
     height: "100vh",
     backgroundColor: "var(--mdc-theme-background)",
-    zIndex: 9, // has to be higher than 5 so it's above advanced settings dialog, and below 10, so the image editor can be displayed above
+    zIndex: 50, // has to be higher than 5 so it's above advanced settings dialog, and below 10, so the image editor can be displayed above
```
Comment (Pavel910) [inline packages/app-admin/src/components/OverlayLayout/OverlayLayout.tsx:16]: @TommyJackson85 one thing to check here: please open the file manager, and then try editing an image, see if image editor is displayed correctly. The comment says `...and below 10, so the image editor can be displayed above`. If image editor is not being displayed, please increase its zIndex as well.

**Comment by TommyJackson85** [comment]
@Pavel Now I am noticing something unusual unlike the last time I went at this project.

I first emptied the page leaving two empty columns.

When I click to drag Image Gallary into a column, its now showing image not found before loading up the Add Images modal. I can select images from the File Manager which loads them into the modal. When I click save on the modal, the error still appears:

**Comment by Pavel910** [comment]
@TommyJackson85 that's not really an error, and it's not related to the issue we're solving in this PR, so just ignore it. 
There's a separate issue for that https://github.com/webiny/webiny-js/issues/1113.

**Comment by TommyJackson85** [comment]
@Pavel910 oh of course, but as you wanted to check what its like to try edit an image, would you think this issue is partially blocked by #1113 ? I thought I had resolved that issue completely! I thought it was just an error with my href link.
Anyway after clicking save on the Image Gallary selected images (from the modal), the editor still seems to be there:
 
Screenshot 2020-08-04 at 19 03 41

**Comment by Pavel910** [comment]
@TommyJackson85 I meant editing an image in the File Manager. So when you click `Add Files` the File Manager shows, and then you can upload an image and edit it (crop, add filters, etc.). Just want to see if zIndex messed up that. Here's what I mean: https://www.webiny.com/static/02-image-editor-137b14eeef10ac5dff4487597765dd43.mp4

**Comment by TommyJackson85** [comment]
Ok I think I found the correct z Index for that and I committed the changes, Image Editor z index is at 51. It appears to work anyway.

**Comment by Pavel910** [comment]
@TommyJackson85 great! I think this PR is now complete, we'll merge it for the next release. 

Good job and thanks! 🍻 🚀

**Comment by TommyJackson85** [comment]
Thanks @Pavel910 ! I learnt something new anyway! 🥇 :)

**Comment by TommyJackson85** [comment]
@Pavel910 just keep in ming I appear to be having "Build & E2E test" fails in this PR.

**Comment by Pavel910** [comment]
@TommyJackson85 yeah we've noticed that. We are working on e2e automated tests, and there are still some glitches we're trying to figure out. We'll resolve it during the week.

**Pull request merged**